frontend/src/store/controller/selectors.ts

```diff
@@ -201,46 +201,45 @@ export const getAlarmLimitsRequestDraft = createSelector(
   getAlarmLimits,
   (alarmLimits: AlarmLimitsRequestResponse): AlarmLimitsRequest | null => alarmLimits.draft,
 );
-const alarmLimitsCurrentSelector = (key: string) =>
-  createSelector(getAlarmLimitsCurrent, (alarmLimitsCurrent: AlarmLimits | null): Range | null =>
-    alarmLimitsCurrent === null
-      ? null
-      : ((alarmLimitsCurrent as unknown) as Record<string, Range>)[key],
+type OptionalAlarmLimitsLike = AlarmLimits | AlarmLimitsRequest | null;
+type AlarmLimitsSelector = OutputSelector<
+  StoreState,
+  AlarmLimits | null,
+  (res: AlarmLimitsRequestResponse) => OptionalAlarmLimitsLike
+>;
+const rangeSelector = (alarmLimitsSelector: AlarmLimitsSelector) =>
+  createSelector(
+    alarmLimitsSelector,
+    (alarmLimits: AlarmLimits | AlarmLimitsRequest | null): Record<string, Range> | null =>
+      alarmLimits === null ? null : ((alarmLimits as unknown) as Record<string, Range>),
   );
 
-export const getAlarmLimitsCurrentSpo2 = alarmLimitsCurrentSelector('spo2');
-export const getAlarmLimitsCurrentHR = alarmLimitsCurrentSelector('hr');
+export const getAlarmLimitsRangeCurrent = rangeSelector(getAlarmLimitsCurrent);
+export const getAlarmLimitsRangeDraft = rangeSelector(getAlarmLimitsRequestDraft);
 
-const alarmLimitsStandbySelector = (key: string) =>
-  createSelector(
-    getAlarmLimitsRequestDraft,
-    (alarmLimitsRequestDraft: AlarmLimitsRequest | null): Range | null =>
-      alarmLimitsRequestDraft === null
-        ? null
-        : ((alarmLimitsRequestDraft as unknown) as Record<string, Range>)[key],
-  );
-export const getAlarmLimitsStandbySpo2 = alarmLimitsStandbySelector('spo2');
-export const getAlarmLimitsStandbyHR = alarmLimitsStandbySelector('hr');
-
-export const getAlarmLimitsUnsavedChanges = createSelector(
-  getAlarmLimitsCurrentSpo2,
-  getAlarmLimitsCurrentHR,
-  getAlarmLimitsStandbySpo2,
-  getAlarmLimitsStandbyHR,
+export const getAlarmLimitsUnsavedKeys = createSelector(
+  getAlarmLimitsRangeCurrent,
+  getAlarmLimitsRangeDraft,
```

Comment:
Here you might as well just pass in `getAlarmLimitsCurrent` and `getAlarmLimitsRequestDraft`, and cast them into `Record<string, Range>` inside the selector. I'd suggest having an `if` check at the start of the selector to return an empty array early if either `getAlarmLimitsCurrent` or `getAlarmLimitsRequestDraft` is null, so that you can take the `currentLimits !== null` and `draftLimits !== null` checks out of the `keys.filter` operation. If both are not null, then you can cast them into `Record<string, Range>`s.

Reply:
yeah I did not realize the difference until now, using getAlarmLimitsRequest now